pkg/cache/v3/simple.go

```diff
@@ -210,6 +211,28 @@ func (cache *snapshotCache) SetSnapshot(node string, snapshot Snapshot) error {
 				delete(info.watches, id)
 			}
 		}
+
+		// this should always pass since we initialize on snapshot creation
+		if snapshot.VersionMap != nil {
+			err := snapshot.ConstructVersionMap()
```

Comment:
Maybe do this only if len(deltaWatches) > 0? That way we don't waste time for sotw computing the hashes

Reply:
Cool I like that